### PR TITLE
(maint) Support testing with v2 inventory in more places

### PR DESCRIPTION
**Pull request by adreyer**
I think these are useful improvements but they don't really fit GH-1191

### CODE REVIEW

File: lib/bolt_spec/run.rb [hunk 1]
```diff
@@ -130,20 +130,31 @@ class BoltRunner
...
-        @inventory = inventory
+      def initialize(config_data, inventory_data, boltdir_path)
+        @config_data = config_data || {}
+        @inventory_data = inventory_data || {}
+        @boltdir_path = boltdir_path
         @analytics = Bolt::Analytics::NoopClient.new
       end
 
+      def config
+        @config = Bolt::Config.new(Bolt::Boltdir.new(@boltdir_path), @config_data)
```
Comment (donoghuc) [inline lib/bolt_spec/run.rb]: Does this need to be recreated every time?

File: lib/bolt_spec/run.rb [hunk 2]
```diff
@@ -130,20 +130,31 @@ class BoltRunner
...
+      end
+
+      def inventory
+        @inventory ||= Bolt::Inventory.create_version(@inventory_data, config, plugins)
+      end
+
+      def plugins
+        @plugins ||= Bolt::Plugin.setup(config, pal, puppetdb_client, @analytics)
```
Comment (donoghuc) [inline lib/bolt_spec/run.rb]: Need to `require 'bolt/plugin` for this constant.

**Comment by donoghuc** [comment]
Seems like `BoltSpec::Run#apply_manifest` was broken (for inventory v1) before this change. I guess it is fairly expensive to integration test that (along with the other run methods). But this appears to work with inventory v1 and v2.

File: lib/bolt_spec/run.rb (hunk 1 above)
Comment (adreyer) [reply lib/bolt_spec/run.rb]: no that was a typo.

**Pull request merged**